flocker/acceptance/endtoend/test_dataset.py

```diff
@@ -102,3 +109,69 @@ def not_exists():
             return deleted
         created.addCallback(delete_dataset)
         return created
+
+    @require_moving_backend
+    @run_test_with(async_runner(timeout=timedelta(minutes=6)))
+    @require_cluster(2)
+    def test_dataset_move_from_dead_node(self, cluster):
+        """
+        A dataset can be moved from one node to another.
+
+        All attributes, including the maximum size, are preserved.
+        """
+        api = get_backend_api(self, cluster.cluster_uuid)
+        if not ICloudAPI.providedBy(api):
+            raise SkipTest(
+                "Backend doesn't support ICloudAPI; therefore it might support"
+                " moving from dead node but as first pass we assume it "
+                "doesn't.")
+
+        # Find a node which is not running the control service.
+        # If the control node is shut down we won't be able to move anything!
+        node = list(node for node in cluster.nodes
+                    if node.public_address !=
+                    cluster.control_node.public_address)[0]
+        other_node = list(other_node for other_node in cluster.nodes
+                          if other_node != node)[0]
+        waiting_for_create = create_dataset(self, cluster, node=node)
+
+        def startup_node(node_id):
+            api.start_node(node_id)
+            # Give node some minimal amount of time to boot so next test
+            # is happier:
+            sleep(20)
```

Comment:
Ah, wait for SSH to come up, which is what I did for reboot test, and that worked well.